libs/image-utils/src/jest_pdf_snapshot.ts

```diff
@@ -0,0 +1,34 @@
+import { iter } from '@votingworks/basics';
+import { readFile } from 'fs/promises';
+import { tmpNameSync } from 'tmp';
+import { Buffer } from 'buffer';
+import { pdfToImages } from './pdf_to_images';
+import { writeImageData } from './image_data';
+
+/**
+ * Custom `jest` matcher to compare a PDF to a snapshot. Accepts a buffer or
+ * path to a PDF file. Converts the PDF to PNG files and uses
+ * `jest-image-snapshot` to snapshot them.
+ */
+export async function toMatchPdfSnapshot(
+  received: string | Buffer
+): Promise<jest.CustomMatcherResult> {
+  const pdfContents =
+    typeof received === 'string' ? await readFile(received) : received;
+  const pdfPages = pdfToImages(pdfContents, { scale: 200 / 72 });
+  iter(pdfPages)
```

Comment:
Nothing here is actually consuming the iterable, so it's effectively a no-op. You could call `toArray` to run the iteration as a side-effect, but this is probably an example of where it's better to use a `for await` than `iter` since you're not using it to generate a value.